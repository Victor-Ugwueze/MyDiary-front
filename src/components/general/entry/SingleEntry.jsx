// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const EntryItem = ({ entry, showEntry }) => (
    <li className='entry-item'>
      <div className=''>
        <h4 className='sing-diary-title diary-text' data-id={entry.id} onClick={showEntry}>{entry.title}</h4>
        <p className='sing-diary-body diary-text'>{entry.body}</p>
      </div>
      <p className='created-at'>{entry.created_at}</p>
      <a className='action'>
        <span data-target='edit-diary-entry' className='btn btn-primary action-edit'>
          <img className='diary-edit icon-edit' src='/images/edit.png' />
          <span className='edit-text'>Edit</span>
        </span>
        &nbsp;|
        <span className='btn btn-danger action-delete'>
          <img className='diary-edit icon-edit' src='/images/delete-button.png' />
          <span className='delete-text'>Delete</span>
        </span>
      </a>
      <div className='arrow-up' />
    </li>
);

EntryItem.propTypes = {
  entry: PropTypes.object,
  showEntry: PropTypes.func,
};

export default EntryItem;
