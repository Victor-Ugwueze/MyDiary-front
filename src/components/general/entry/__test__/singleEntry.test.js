// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// components
import SingleEntry from '../SingleEntry';


configure({
  adapter: new Adapter(),
});

describe('Renders the single itme component', () => {
  it('should render SingleEntry', () => {
    const component = shallow(
      <SingleEntry entry={{ id: 1, title: 'title' }}/>
    );
    expect(component).toMatchSnapshot();
  });
});
