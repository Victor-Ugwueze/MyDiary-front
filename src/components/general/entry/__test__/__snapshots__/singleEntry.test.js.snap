// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the single itme component should render SingleEntry 1`] = `
<li
  className="entry-item"
>
  <div
    className=""
  >
    <h4
      className="sing-diary-title diary-text"
      data-id={1}
    >
      title
    </h4>
    <p
      className="sing-diary-body diary-text"
    />
  </div>
  <p
    className="created-at"
  />
  <a
    className="action"
  >
    <span
      className="btn btn-primary action-edit"
      data-target="edit-diary-entry"
    >
      <img
        className="diary-edit icon-edit"
        data-id={1}
        data-mode="edit"
        src="/images/edit.png"
      />
      <span
        className="edit-text"
        data-id={1}
        data-mode="edit"
      >
        Edit
      </span>
    </span>
    Â |
    <span
      className="btn btn-danger action-delete"
    >
      <img
        className="diary-edit icon-edit"
        data-id={1}
        data-mode="initiate"
        src="/images/delete-button.png"
      />
      <span
        className="delete-text"
        data-id={1}
        data-mode="initiate"
      >
        Delete
      </span>
    </span>
  </a>
  <div
    className="arrow-up"
  />
</li>
`;
